### DISCUSSION TITLE
❓ Questions / Help / Support

### DISCUSSION THREAD

**Discussion by eliran-fm** [snakers4/silero-vad#319] [Q&A] [answered]
## `max_speech_duration_s` behavior
Hey,
In your documentation you specify the `max_speech_duration_s` parameter, and state that chunks longer than that will be attempted to be split at the last silence that lasts more than 100s, did you mean 100ms? Because in the code I can see the following:
`min_silence_samples_at_max_speech = sampling_rate * 98 / 1000`

Thanks

**Comment by snakers4** [comment]
> min_silence_samples_at_max_speech = sampling_rate * 98 / 1000

Hi, this parameter is used here https://github.com/snakers4/silero-vad/blob/master/utils_vad.py#L317 for a different reason

**Reply by eliran-fm** [reply]
@snakers4 what do you mean by a different reason? If I understand correctly, at [ https://github.com/snakers4/silero-vad/blob/master/utils_vad.py#L317](url) you check if there's enough silence so you can cut the chuck smoothly. My only question is whether in your documentation you meant 100 milliseconds instead of 100 seconds

**Reply by snakers4** [reply]
If you speak about `max_speech_duration_s`, it is 100 seconds in the docsting (as an example), and by default the parameter is set to infinity. 

But I do not understand how this line `min_silence_samples_at_max_speech = sampling_rate * 98 / 1000` is related to this question.

One is min silence samples, the other is max speech duration.

**Reply by eliran-fm** [reply]
there are two parameters - `min_silence_samples_at_max_speech` and `min_silence_samples`.
I figured `min_silence_samples_at_max_speech` has to do with the silence length required for smooth cut when reaching the `max_speech_duration_s`.

For example if I set `max_speech_duration_s` to 30s, I expect to get chunks shorter than 30s, but also need some parameter to set the silence required for non-aggressive cut, which from the doc I understand is hardcoded to 100s (and I suspect you meant 100ms instead)
Did I misunderstood?

**Reply by snakers4** [reply]
> but also need some parameter to set the silence required for non-aggressive cut, which from the doc I understand is hardcoded to 100s (and I suspect you meant 100ms instead)

I cannot see such parameter in the code, can you please point to it?

**Reply by eliran-fm** [reply]
My understanding was that `min_silence_samples_at_max_speech` is exactly for that (the name also states that)

From the doc:
`Chunks longer than max_speech_duration_s will be split at the timestamp of the last silence that lasts more than 100s (if any), to prevent agressive cutting.`
Meaning there's the `max_speech_duration_s` which determines the maximum output chunk duration, and some variable (which I presume is `min_silence_samples_at_max_speech`) that determines the aforementioned "last silence" duration, and is currently fixed to 98 milliseconds

**Reply by snakers4** [reply]
> min_silence_samples_at_max_speech = sampling_rate * 98 / 1000
> which from the doc I understand is hardcoded to 100s 

This is about 100ms, not 100 seconds

**Reply by eliran-fm** [reply, marked as answer]
Ok I see, so maybe it's worth updating the doc
Thank you for clarifying @snakers4 !

**Reply by dayfinihzy** [reply]
Where can I find the doc?
